app/controllers/admin/users_controller.rb

```diff
@@ -361,9 +361,11 @@ def unsilence
   def disable_second_factor
     guardian.ensure_can_disable_second_factor!(@user)
     user_second_factor = @user.user_second_factors
-    raise Discourse::InvalidParameters unless !user_second_factor.empty?
+    user_security_key = @user.security_keys
+    raise Discourse::InvalidParameters if user_second_factor.empty? && user_security_key.empty?
```

Comment:
Is this check necessary? What if the DB is in a state where the security keys are already gone, should we raise an error then?

Reply:
That's what this is, right? Raise an error if there's nothing to delete.

Reply:
Sorry, I thought the issue was that the security keys were not being deleted when this happens?

Reply:
The issue was that if there were no TOTP factors but were security keys, the "Remove 2FA" button didn't do anything and generated an error. I need to also fix the client side, actually.